operators/pkg/instance-controller/container_logic.go

```diff
@@ -256,6 +278,24 @@ func (r *InstanceReconciler) CreateContainerEnvironment(
 		return err
 	}
 
+	if !environment.Resources.Disk.IsZero() {
+		pvc := v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+			Spec: buildContainerInstancePVCSpec(environment),
+		}
+
+		if _, err := ctrl.CreateOrUpdate(ctx, r.Client, &pvc, func() error {
+			// PVC's specs are immutable, they have to be set directly
+			return ctrl.SetControllerReference(instance, &pvc, r.Scheme)
+		}); err != nil {
+			r.setInstanceStatus(ctx, "Could not create PVC "+pvc.Name+" in namespace "+pvc.Namespace+": "+err.Error(), "Error", "VmiNotCreated", instance, "", "")
+			return err
+		}
```

Comment:
Please, add a log line also in case of success, as [here](https://github.com/netgroup-polito/CrownLabs/blob/7c77875eca40d237e32f86b625a00a33c6c1e3a4/operators/pkg/tenant-controller/workspace_controller.go#L276), to simplify troubleshooting